### PR TITLE
FieldDef type

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

File: src/compiler/legend.ts [hunk 1]
```diff
@@ -43,7 +43,7 @@ export function compileLegend(model: Model, channel: Channel, def) {
   });
 
   // 2) Add mark property definition groups
-  let props = legend.properties || {};
+  const props = typeof legend === 'Legend' && legend.properties || {};
```
Comment (kanitw) [inline src/compiler/legend.ts]: This line is terrible!  Nobody on earth remember precedence of these two operators. 
I also do not get what you try to do here?

File: src/schema/sort.schema.ts [hunk 2]
```diff
@@ -8,6 +8,11 @@ export interface Sort {
   order?: string;
 }
 
+export function isSort(object: any): object is Sort {
```
Comment (kanitw) [inline src/schema/sort.schema.ts]: I don't understand why do you need to add this?

Comment (domoritz) [inline src/schema/sort.schema.ts]: You cannot do runtime type checking.

Comment (kanitw) [inline src/schema/sort.schema.ts]: First, I still do not get it. 
Second, if there is no way to understand this easily, should we ignore it?

Comment (domoritz) [inline src/schema/sort.schema.ts]: A good explanation is https://github.com/Microsoft/TypeScript/issues/1007

File: src/compiler/legend.ts [hunk 3]
```diff
@@ -58,9 +60,12 @@ export function compileLegend(model: Model, channel: Channel, def) {
 }
 
 export function title(model: Model, channel: Channel) {
-  let leg = model.fieldDef(channel).legend;
+  // https://github.com/Microsoft/TypeScript/issues/5842
+  const legend: any = model.fieldDef(channel).legend;
 
-  if (leg.title) return leg.title;
+  if (isLegend(legend) && legend.title) {
```
Comment (kanitw) [inline src/compiler/legend.ts]: What's the problem here?

Comment (domoritz) [inline src/compiler/legend.ts]: `legend` is of type `Legend | boolean` so it does not necessarily have a property `title`.

Comment (kanitw) [inline src/compiler/legend.ts]: But if there is no property title, this predicate is then false, so that's fine?

Comment (domoritz) [inline src/compiler/legend.ts]: Yes, but not for the type checker. 

```js
var a = true;
a.title  // you expect this to crash
```

**Pull request merged**